### ISSUE TITLE
Openshift 4.3.8 -- x509: certificate has expired or is not yet valid

### ISSUE THREAD

**Issue by Pooriya-a** [openshift/installer#3502]
I am trying to stand up an Openshift 4.3 cluster using the following machines. Bootstrap node successfully gets installed and then I start the master nodes. The master node gets installed with RHELCOS and then reboots, but at the reboot time it shows the following error on the master console. I have also attached the log bundle. 

```
ignition[798] Get error: Get https://api-int.ocp4.contoso.com:22263/config/master: x509: certificate has expired or is not yet valid.
``

Basion= running (dns, dhcp, tftp, haproxy)
bootstrap
3 masters
4 workers

 I have ensured that the DNS is configured correctly and name resolution is happening correctly. When I curl into "curl -vv api-int.ocp4.contoso.com:22623", I receive the following out put.

---------------
* Rebuilt URL to: api-int.ocp4.contoso.com:22623/
*   Trying 172.16.1.10...
* TCP_NODELAY set
* Connected to api-int.ocp4.contoso.com (172.16.1.10) port 22623 (#0)
> GET / HTTP/1.1
> Host: api-int.ocp4.contoso.com:22623
> User-Agent: curl/7.61.1
> Accept: */*
>
* HTTP 1.0, assume close after body
< HTTP/1.0 400 Bad Request
<
Client sent an HTTP request to an HTTPS server.
* Closing connection 0

----------------
[log-bundle-20200423235736.zip](https://github.com/openshift/installer/files/4526661/log-bundle-20200423235736.zip)
![Capture](https://user-images.githubusercontent.com/29644478/80175233-df448b80-85a9-11ea-8f91-3b6e6de31dc3.PNG)




# Version

```console
$ openshift-install version
./openshift-install 4.3.8
built from commit f7a2f7cf9ec3201bb8c9ebb677c05d21c72e3cc5
release image quay.io/openshift-release-dev/ocp-release@sha256:a414f6308db72f88e9d2e95018f0cc4db71c6b12b2ec0f44587488f0a16efc42
```

# Platform:

# What happened?
Following is the log from approve-csr.log
```
-- Logs begin at Thu 2020-04-23 23:47:55 UTC, end at Thu 2020-04-23 23:57:36 UTC. --
Apr 23 23:49:55 bootstrap.ocp4.contoso.com systemd[1]: Started Approve CSRs during bootstrap phase.
Apr 23 23:49:55 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: Approving all CSR requests until bootstrapping is complete...
Apr 23 23:49:56 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:50:16 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:50:36 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:50:56 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:51:16 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:51:36 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:51:57 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:52:17 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:52:37 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:52:57 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:53:17 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: The connection to the server localhost:6443 was refused - did you specify the right host or port?
Apr 23 23:53:37 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: error: the server doesn't have a resource type "csr"
Apr 23 23:53:57 bootstrap.ocp4.contoso.com approve-csr.sh[2291]: error: the server doesn't have a resource type "csr"
```

Following is the log from bootkube.log

Apr 23 23:52:49 bootstrap.ocp4.contoso.com bootkube.sh[2288]: Rendering CCO manifests...
Apr 23 23:53:12 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:12.719216007 +0000 UTC m=+22.943947290 image pull  
Apr 23 23:53:12 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:12.73412813 +0000 UTC m=+22.958859371 container create f1acf14bdf601dbd6761ac43a5df2c2330a7b8d3d530a5bc5e0f33f40c1ab15e (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e15e7a730451443a0829b11e8f82bcc11d1ab09a954a1c77d1223470466020bd, name=pedantic_maxwell)
Apr 23 23:53:13 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:13.017773283 +0000 UTC m=+23.242504565 container init f1acf14bdf601dbd6761ac43a5df2c2330a7b8d3d530a5bc5e0f33f40c1ab15e (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e15e7a730451443a0829b11e8f82bcc11d1ab09a954a1c77d1223470466020bd, name=pedantic_maxwell)
Apr 23 23:53:13 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:13.038219131 +0000 UTC m=+23.262950373 container start f1acf14bdf601dbd6761ac43a5df2c2330a7b8d3d530a5bc5e0f33f40c1ab15e (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e15e7a730451443a0829b11e8f82bcc11d1ab09a954a1c77d1223470466020bd, name=pedantic_maxwell)
Apr 23 23:53:13 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:13.038294459 +0000 UTC m=+23.263025692 container attach f1acf14bdf601dbd6761ac43a5df2c2330a7b8d3d530a5bc5e0f33f40c1ab15e (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e15e7a730451443a0829b11e8f82bcc11d1ab09a954a1c77d1223470466020bd, name=pedantic_maxwell)
Apr 23 23:53:13 bootstrap.ocp4.contoso.com bootkube.sh[2288]: time="2020-04-23T23:53:13Z" level=info msg="Rendering files to /assets/cco-bootstrap"
Apr 23 23:53:13 bootstrap.ocp4.contoso.com podman[4469]: 2020-04-23 23:53:13.094996733 +0000 UTC m=+23.319727974 container died f1acf14bdf601dbd6761ac43a5df2c2330a7b8d3d530a5bc5e0f33f40c1ab15e (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e15e7a730451443a0829b11e8f82bcc11d1ab09a954a1c77d1223470466020bd, name=pedantic_maxwell)
Apr 23 23:53:13 bootstrap.ocp4.contoso.com bootkube.sh[2288]: Starting etcd certificate signer...
Apr 23 23:53:27 bootstrap.ocp4.contoso.com podman[4744]: 2020-04-23 23:53:27.109887959 +0000 UTC m=+13.996250617 image pull  
Apr 23 23:53:27 bootstrap.ocp4.contoso.com podman[4744]: 2020-04-23 23:53:27.124965764 +0000 UTC m=+14.011328415 container create b19a39f082b28b7d21f71e5340d729812be53d04a74b0d06dac2b95ba6db97ed (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9160d52b7b0ddb43679b489f04ab99ca8ee56b98a14e4d1a35564d22c3ae76cc, name=etcd-signer)
Apr 23 23:53:27 bootstrap.ocp4.contoso.com podman[4744]: 2020-04-23 23:53:27.428007824 +0000 UTC m=+14.314370476 container init b19a39f082b28b7d21f71e5340d729812be53d04a74b0d06dac2b95ba6db97ed (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9160d52b7b0ddb43679b489f04ab99ca8ee56b98a14e4d1a35564d22c3ae76cc, name=etcd-signer)
Apr 23 23:53:27 bootstrap.ocp4.contoso.com podman[4744]: 2020-04-23 23:53:27.439753962 +0000 UTC m=+14.326116613 container start b19a39f082b28b7d21f71e5340d729812be53d04a74b0d06dac2b95ba6db97ed (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9160d52b7b0ddb43679b489f04ab99ca8ee56b98a14e4d1a35564d22c3ae76cc, name=etcd-signer)
Apr 23 23:53:27 bootstrap.ocp4.contoso.com bootkube.sh[2288]: b19a39f082b28b7d21f71e5340d729812be53d04a74b0d06dac2b95ba6db97ed
Apr 23 23:53:27 bootstrap.ocp4.contoso.com bootkube.sh[2288]: Waiting for etcd cluster...
Apr 23 23:53:51 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:51.619603418 +0000 UTC m=+24.162943662 image pull  
Apr 23 23:53:51 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:51.632825641 +0000 UTC m=+24.176165853 container create 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:51 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:51.900992891 +0000 UTC m=+24.444333123 container init 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:51 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:51.920688181 +0000 UTC m=+24.464028385 container start 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:51 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:51.920760801 +0000 UTC m=+24.464101003 container attach 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: {"level":"warn","ts":"2020-04-23T23:53:56.938Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-1de76475-cc6c-4304-b302-c5a283a47bff/etcd-2.ocp4.contoso.com:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 172.16.1.22:2379: connect: no route to host\""}
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: {"level":"warn","ts":"2020-04-23T23:53:56.938Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-00e5b0f2-ea29-4e53-a3c9-ce01c69b0b1d/etcd-1.ocp4.contoso.com:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 172.16.1.21:2379: connect: no route to host\""}
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: {"level":"warn","ts":"2020-04-23T23:53:56.938Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-288d91a1-5559-48be-a4c2-8632018bffea/etcd-0.ocp4.contoso.com:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 172.16.1.20:2379: connect: connection refused\""}
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: https://etcd-2.ocp4.contoso.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: https://etcd-1.ocp4.contoso.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: https://etcd-0.ocp4.contoso.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
Apr 23 23:53:56 bootstrap.ocp4.contoso.com bootkube.sh[2288]: Error: unhealthy cluster
Apr 23 23:53:56 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:56.983334987 +0000 UTC m=+29.526675215 container died 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:57 bootstrap.ocp4.contoso.com podman[4833]: 2020-04-23 23:53:57.052297691 +0000 UTC m=+29.595637933 container remove 26ec972499dbcf93dcf9f9d6c7655e06568125d8755f72d582303bb4734e889a (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:53:57 bootstrap.ocp4.contoso.com bootkube.sh[2288]: etcdctl failed. Retrying in 5 seconds...
Apr 23 23:54:02 bootstrap.ocp4.contoso.com podman[4967]: 2020-04-23 23:54:02.184548403 +0000 UTC m=+0.116139596 container create 94fbfb2b531c107d86c08a225e2bc71a87b297995bf8f7dc7e56ea37392e99ae (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:54:02 bootstrap.ocp4.contoso.com podman[4967]: 2020-04-23 23:54:02.430144404 +0000 UTC m=+0.361735603 container init 94fbfb2b531c107d86c08a225e2bc71a87b297995bf8f7dc7e56ea37392e99ae (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:54:02 bootstrap.ocp4.contoso.com podman[4967]: 2020-04-23 23:54:02.443327094 +0000 UTC m=+0.374918285 container start 94fbfb2b531c107d86c08a225e2bc71a87b297995bf8f7dc7e56ea37392e99ae (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)
Apr 23 23:54:02 bootstrap.ocp4.contoso.com podman[4967]: 2020-04-23 23:54:02.44346274 +0000 UTC m=+0.375053943 container attach 94fbfb2b531c107d86c08a225e2bc71a87b297995bf8f7dc7e56ea37392e99ae (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9a7267ad2a1146abdfd69eff5a9ccb5cb1c4b06ad92404b97fa34ecebd31a05, name=etcdctl)

# What you expected to happen?
I have been expecting that the bootstrap node bootstrap the masters and build the control plane, but it has not been successful. I have created all the machines using VMware workstation and have followed all the steps in the following two links to stand the cluster up.

https://www.techbeatly.com/2019/07/openshift-4-libvirt-upi-installation.html#.Xm3r50oTFhF
https://www.openshift.com/blog/openshift-4-bare-metal-install-quickstart

# Anything else we need to know?

Enter text here.

# References

- enter text here.

**Comment by Pooriya-a** [comment]
I had also validated the certificate and the certificate is valid as shown below.

</dev/null openssl s_client -showcerts -connect api-int.ocp4.contoso.com:22623 | openssl x509 -in /dev/stdin -noout -text

depth=0 CN = api-int.ocp4.contoso.com
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = api-int.ocp4.contoso.com
verify error:num=21:unable to verify the first certificate
verify return:1
DONE
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 6239443789137359926 (0x5696f511778cb436)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: OU = openshift, CN = root-ca
        Validity
            Not Before: Apr 24 04:00:36 2020 GMT
            Not After : Apr 22 04:00:37 2030 GMT
        Subject: CN = api-int.ocp4.contoso.com
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:d2:7b:ae:c9:7f:a8:d2:e6:7b:09:0d:28:4e:b1:
                    e3:ec:70:79:94:4d:c5:73:af:fb:2a:96:a1:97:b4:
                    53:f4:12:1f:83:39:cc:bf:27:5e:34:e0:56:92:b4:
                    02:36:70:17:1d:d1:19:2d:64:54:b6:de:db:f2:b0:
                    7a:a7:6d:ac:00:7a:f4:02:62:f6:32:23:8c:04:ac:
                    ca:ce:07:65:90:96:e6:2c:f4:a9:47:64:1d:75:38:
                    4e:d5:77:67:ae:db:da:ea:81:d7:31:c1:04:0f:d1:
                    64:b5:01:8e:c2:fa:b6:90:3f:32:f5:ef:94:4d:37:
                    65:3a:6c:51:64:73:f7:da:50:49:08:f4:a8:3c:a7:
                    88:96:20:ff:8a:03:dc:b3:e4:38:51:7c:46:21:b4:
                    6b:60:8a:f1:36:47:2e:a0:87:c8:dc:39:60:11:9c:
                    b8:87:be:48:55:a5:9c:99:70:fb:a2:7f:1d:0b:6e:
                    2f:88:7c:00:63:ca:fe:51:8c:f8:38:e2:21:6d:42:
                    d1:06:60:0a:dd:31:2f:6f:7b:8d:5e:30:f8:b0:fc:
                    9c:29:6d:eb:7e:4b:c6:69:8f:97:5b:52:44:58:82:
                    ea:cf:9a:42:6b:1c:ef:28:9a:9f:23:3d:10:5b:b2:
                    99:f2:2c:af:a1:be:f9:c4:dd:13:5a:61:83:16:cf:
                    87:41
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Extended Key Usage:
                TLS Web Server Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier:
                58:60:60:D4:15:49:4D:5F:15:B2:15:F3:E3:A9:17:08:0C:06:B9:61
            X509v3 Authority Key Identifier:
                keyid:58:60:60:D4:15:49:4D:5F:15:B2:15:F3:E3:A9:17:08:0C:06:B9:61

            X509v3 Subject Alternative Name:
                DNS:api-int.ocp4.contoso.com
    Signature Algorithm: sha256WithRSAEncryption
         9e:b2:ce:62:b1:9f:e8:7e:73:5c:c5:96:c2:5e:ee:9f:d0:34:
         31:5c:df:8c:ec:00:d6:f8:bd:3c:56:1a:bc:b0:be:dd:08:7e:
         4e:13:59:b3:0f:81:24:9b:7c:19:c4:46:f3:ce:c4:39:96:6a:
         40:98:04:06:56:a8:7c:20:20:42:fe:da:ba:3b:19:5e:7e:2d:
         1c:4d:5c:11:c7:6d:87:55:84:c3:85:f4:1f:f5:26:38:84:38:
         58:3a:4b:36:ab:d0:00:5c:74:72:69:28:5e:12:a9:e0:05:c9:
         17:5e:99:d0:ed:dc:0a:b1:5d:86:f9:a2:f0:30:39:43:76:0d:
         13:78:ce:7f:aa:7a:ba:a6:18:30:6e:23:f6:38:6e:f5:92:18:
         1c:59:e9:fa:03:44:e4:35:e2:b6:1b:04:24:d5:a1:2b:4e:88:
         1d:35:61:1a:7a:f0:3f:e5:1f:10:13:cf:78:c5:c3:7e:4c:5a:
         52:e8:7f:89:f1:8f:a0:4a:fd:03:99:10:d8:66:6b:9c:59:70:
         21:8d:35:eb:37:9f:fa:7b:e0:78:d2:a0:cc:c4:8c:4f:71:e8:
         1a:18:f8:69:b6:0a:d3:34:eb:9d:83:0b:49:73:85:11:e8:aa:
         45:76:fc:0e:69:b4:f5:24:4f:43:52:b9:70:1e:7e:95:5f:e7:
         ca:41:6b:b9

**Comment by abhinavdahiya** [comment]
> Get https://api-int.ocp4.contoso.com:22263/config/master: x509: certificate has expired or is not yet valid

Is seems like the machine's clock is skewed enough that the certs generated are not yet valid.

I think you'll have to make sure your machine has some initial time sync to accept the certs correctly.

**Comment by jcpowermac** [comment]
@Pooriya-a can you make sure that the machine that you are running the installer from and all of your ESXi hosts are configured and running NTP?

Also make `.openshift_install_state.json` is deleted before each install attempt.

See: https://docs.openshift.com/container-platform/4.3/installing/installing_vsphere/installing-vsphere.html#installation-vsphere-infrastructure_installing-vsphere
at the bottom of that section has the link to VMware's documentation about NTP.

**Comment by Pooriya-a** [comment]
I am not using VMware vSphere. I am using VMware Workstation on my laptop. NTP is enabled on the provisioning machine and I delete the installation folder and ignition files before retrying.

**Comment by Pooriya-a** [comment]
I have checked the time is right on my machine and I even configured chrony to configure ntp on master nodes at the install time.

**Comment by jcpowermac** [comment]
@Pooriya-a 
We don't support workstations afaik.  In any case I haven't tried to install OpenShift in Workstation and I don't have it installed.

Can you look at the virtual machine bios and check the time there - in ESXi its based on the host's time.  Certificates are always in GMT and ESXi is always UTC.  I am not sure what workstation does.

**Comment by Pooriya-a** [comment]
What do you suggest me to set the time in my bastion machine or ntp server please? The time in bios is set to my local time zone and is correct. Do you recommend setting the time to GMT in my provisioning machine please?

**Comment by jcpowermac** [comment]
Maybe set everything to GMT/UTC temporarily and see if it works.

**Comment by Pooriya-a** [comment]
I tried UTC and it did not.

**Comment by jcpowermac** [comment]
@Pooriya-a any luck?

**Comment by Pooriya-a** [comment]
I managed to make it working by using just the same configuration (DNS, TFTP, HAPROXY, etc) in Ubuntu using Libvirt and in Windows using Virtualbox. When using VMware workation, my bootstrap node gets deployed successfully, but my master nodes can't be deployed throwing the certificate error that I provided above. I tried using chrony to configure time in master node using machine config, I tried settings my bastion node time to UTC, GMT, and local time. I also tried configuring ntp server information using DHCP server, but this configuration did not work in VMware Workstation. I also checked the time in master nodes' BIOS and the time was correct. I am not sure why this happens, but this only happened in VMware Workstation.
